frontend/src/store/my-loans/actions.ts

```diff
@@ -10,6 +13,21 @@ const actions = {
         loans
       })
     })
+  },
+  [TRANSFER_COLLATERAL]({ commit }, payload: { shortId: string, amount: string, onSuccess: () => void }) {
+    (async function () {
+      const { shortId, amount, onSuccess } = payload;
+      commit('START_TRANSFERING_COLLATERAL', { shortId });
+      const oldBalance: number = await getMyBalance();
+      await gatherCollateral(shortId, amount);
```

Comment:
Your error checking here won't work. If transferring the collateral fails, this await will throw an Error. You should probably do something like:

```typescript
gatherCollateral(shortId, amount).then(()=>{
   commit('COLLATERAL_TRANSFERED', { shortId });
   onSuccess();
}).catch((err: Error)=>{
  console.error(err);
  commit('SOME_GENERIC_ERROR_PAGE', { msg: "Transferring collateral failed. Do you have enough funds?" });
});
```